source/gui.py

```diff
@@ -0,0 +1,332 @@
+                self.in_hand.append(pallet_item)
+                if (self.verbose_events): print (f'Picked up {pallet_item.name}')
+                return
+            
+            # Pick_up event did not happen in the pallet
+            event.x -= self.pallet_item_size().x
+            for shape in self.list_of_canvas_shapes:
+                if event.x < shape.box.pos.x: continue                      # Left of shape.box  (out of range)
+                if event.x > shape.box.pos.x + shape.box.size.x: continue   # Right of shape.box (out of range)
+                if event.y < shape.box.pos.y: continue                      # Top of shape.box   (out of range)
+                if event.y > shape.box.pos.y + shape.box.size.y: continue   # Bottom of shape.box(out of range)
+
+                self.in_hand.append(shape)
+                shape.remove_shape(self)
+                self.list_of_canvas_shapes.remove(shape)
+                if (self.verbose_events): print (f'Shape picked up from: {shape.box}')
+                return
+            if (self.verbose_events): print ('No shape in the region')
+        def let_go(event):
+            if (self.verbose_events): print(f'<let_go> at {event.x},{event.y}')
+            pallet_item = get_pallet_item(loc.Pos(event.x, event.y))
+
+            if pallet_item != PalletItem.NONE:
+                print (f"Tossed: {len(self.in_hand)} item's in the garbage can.")
+                self.in_hand.clear()
+                return
+            
+            # Let go on the canvas
+            event.x -= self.pallet_item_size().x
+            for item in self.in_hand:
+                # Relocate the shape        
+                if not isinstance(item, PalletItem):
+                    rotation_rad = 0.0
+                    depth_percentage = 0
+                    try:
+                        rotation_rad = item.rotation_rad
+                        depth_percentage = item.depth_percentage
+                    except: pass
+                    new_shape = get_new_shape(shape=PalletItem(int(item.class_id)+PalletItem.CIRCLE.value),
+                                              center_pos=loc.Pos(event.x, event.y),
+                                              size=item.box.size,
+                                              color=PalletItem[item.fill_color.upper()],
+                                              rotation_rad=rotation_rad,
+                                              depth_percentage=depth_percentage)
+                    self.list_of_canvas_shapes.append(new_shape)
+                    new_shape.draw_shape(tkinter_canvas=self,
+                                         location_offset=loc.Pos(x=self.pallet_item_size().x,y=0))
+                    self.in_hand.remove(item)
+                    continue
+                
+                # Data for new shape
+                new_shape_color = self.last_color.name.lower()
+                new_shape_shape = ''        # ILLEGAL
```

Comment:
should this be in if it is 'ILLEGAL' ?

Reply:
@Maucee, Yea... I guess it was a bit faqe, but i wanted to declare it at that depth.  
replaced it by `# Declared here as empty to be filled in later`